src/test/deploy/functions/containerCleaner.spec.ts

```diff
@@ -209,3 +209,212 @@ describe("ContainerRegistryCleaner", () => {
     expect(stub.rm).to.not.have.been.called;
   });
 });
+
+describe("listGcfPaths", () => {
+  const LOCATIONS_US = Promise.resolve({
+    children: ["us-central1", "us-west2"],
+    digests: [],
+    tags: [],
+  });
+
+  const LOCATIONS_EU = Promise.resolve({
+    children: ["europe-west1", "europe-central2"],
+    digests: [],
+    tags: [],
+  });
+
+  const LOCATIONS_ASIA = Promise.resolve({
+    children: ["asia-northeast1", "asia-south1"],
+    digests: [],
+    tags: [],
+  });
+
+  const EMPTY = Promise.resolve({
+    children: [],
+    digests: [],
+    tags: [],
+  });
+
+  it("should throw an error on invalid location", () => {
+    expect(() => containerCleaner.listGcfPaths("project", ["invalid"])).to.throw;
```

Comment:
Nit: checking for it throwing is fragile because it only passes if the error is thrown before any await command. expect it to reject instead.

Reply:
Fixed